pkg/farm/farm.go

```diff
@@ -0,0 +1,503 @@
+package farm
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/containers/buildah/define"
+	"github.com/containers/common/libimage"
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/containers/podman/v4/pkg/domain/infra"
+	"github.com/containers/storage"
+	"github.com/hashicorp/go-multierror"
+	"github.com/sirupsen/logrus"
+)
+
+// Farm represents a group of connections to builders.
+type Farm struct {
+	name         string
+	storeOptions *storage.StoreOptions           // not nil -> use local engine, too
+	builders     map[string]entities.ImageEngine // name -> builder
+}
+
+// Schedule is a description of where and how we'll do builds.
+type Schedule struct {
+	platformBuilders map[string]string // target->connection
+}
+
+func newFarmWithBuilders(_ context.Context, name string, destinations *map[string]config.Destination, storeOptions *storage.StoreOptions, local bool) (*Farm, error) {
+	farm := &Farm{
+		builders:     make(map[string]entities.ImageEngine),
+		storeOptions: storeOptions,
+		name:         name,
+	}
+	var (
+		builderMutex sync.Mutex
+		builderGroup multierror.Group
+	)
+	// Set up the remote connections to handle the builds
+	for name, dest := range *destinations {
+		name, dest := name, dest
+		builderGroup.Go(func() error {
+			fmt.Printf("Connecting to %q\n", name)
```

Comment:
We've got a lot of these that are carried over from the PoC.  We're sure we still want them?

Reply:
Yes, I left them some of them in there on purpose as it lets the user know what is going on.